### ISSUE TITLE
`./x.py --stage 1 test src/libcore --no-doc` broken

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rust#52943]
It says
```
Testing core stage1 (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
   Compiling libc v0.2.42
   Compiling rand v0.4.2
   Compiling core v0.0.0 (file:///home/r/src/rust/rustc.2/src/libcore)
error: trait objects without an explicit `dyn` are deprecated
  --> libcore/../libcore/benches/any.rs:18:36
   |
18 |         let mut y = &mut x as &mut Any;
   |                                    ^^^ help: use `dyn`: `dyn Any`
   |
   = note: requested on the command line with `-D bare-trait-objects`

error: aborting due to previous error

error: Could not compile `core`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```

Might be caused by https://github.com/rust-lang/rust/commit/e098985939a1ac5baef4314aee44efc9dbf1b82? Cc @ishitatsuyuki

**Comment by ishitatsuyuki** [comment]
Yeah we should fix it. ... but how did this slip through? Because it's a bench?

**Comment by RalfJung** [comment]
I am running `./x.py test` though, not asking for benches. Strange.

**Comment by ishitatsuyuki** [comment]
It seems I have several things to fix:

- Missing dyns that are not caught by the CI
- rustbuild running benches without them being requested
- Show the dyn warnings when `deny-warnings = false` in config.toml

**Comment by RalfJung** [comment]
It seems the default is to deny?
```
# Whether to deny warnings in crates
#deny-warnings = true
```
But thanks for the hint, I should disable that. ;)

**Comment by ishitatsuyuki** [comment]
@RalfJung I mean, if you set `deny-warnings = false` it doesn't pass the `-W bare-trait-objects` flag.

**Comment by RalfJung** [comment]
I cannot reproduce this any more.
